src/ert/scheduler/scheduler.py

```diff
@@ -263,6 +263,8 @@ async def execute(
         try:
             await self._monitor_and_handle_tasks(scheduling_tasks)
         finally:
+            while not self._events.empty():
```

Comment:
Maybe it is `_monitor_and_handle_tasks` that should not return until this is empty. 
And inside there one might consider if this should be a test inside `is_active()`, that scheduler should consider itself active while there are events pending.

Reply:
Let's give it a try then

Reply:
simulation_context is using `is_active()` so maybe we should put the guard inside there to avoid the same problem popping up there.

Reply:
I vote for changing `is_active()` to return True whenever the event queue is nonempty.

Reply:
Getting there, I see we need to mock `websocket.client.connect` in order to test this.

Reply:
And possibly we should only require the queue to become empty if `self.__ee_uri` is not None.

Reply:
I needed to change it back to a version where task handles cancellation itself. Two main reasons:
1) The way task is doing cleanup should be an implementation detail and as such we don't want to create class variables (eg. publisher running) and maintain them.
2) Simulation context handling becomes far more complex just to check publisher running and queue not empty - I mean the tests mostly.
Thus the best option would be to do the cleanup once we cancel the task as this would become an implementation detail of the publisher task.